suno-factory/backend/src/api/v1/endpoints/generate.py

```
@@ -0,0 +1,31 @@
+@router.post("/process", response_model=SongPackage)
+async def generate_song_package(
+    file: UploadFile = File(...),
+    instructions: str | None = Form(None),
+) -> SongPackage:
+    """
+    Upload -> Parse -> Compose -> Return JSON package.
+    """
+    try:
+        raw_text = await FileParser.process_upload(file)
+    except HTTPException as exc:
+        raise exc
```

Ruff (0.15.1)

[warning] Use `raise` without specifying exception name

Remove exception name

(TRY201)